content/docs/ref/runner.md

```diff
@@ -99,3 +285,57 @@ Any [generic option](/doc/ref) in addition to:
 
    replacing the `IP_ADDRESS` placeholder with the instance address returned by
    `cml runner` (search the output logs for `instanceIp`).
+
+### Using `--cloud-startup-script`
+
+Provide a Base64 encoded bash script that will be executed as part of the instance provisioning for `cml runner`. Your script will be run after `cml runner` does it's initial setup but before your runner becomes available to the CI/CD provider.
+
+<admon type="info">
+
+The execution of this script counts towards the "provisioning time" of the `cml runner` command, as such if it(total time) takes longer that 10 minutes, then provisioning is considered failled, the instance will be deleted, and the CI/CD job will exit with a failed state.
+
+</admon>
+
+### Basic Usage
+The simplest example can be seen as this:
+```
+cml runner ...
+  --cloud-startup-script=IyEvYmluL2Jhc2ggCgplY2hvICJoZWxsbyB3b3JsZCIK \
+  ...
+```
+where `IyEvYmluL2Jhc2ggCgplY2hvICJoZWxsbyB3b3JsZCK` is:
+```bash
+#!/bin/bash
+
+echo "hello world"
+```
+
+### Debugging example
+For a more useful example to help you debug your job or `cml runner` see below, which will add you ssh keys the cml instance.
+```
+cml runner ...
+  --cloud-startup-script=$(echo 'echo "$(curl https://github.com/'"$GITHUB_ACTOR"'.keys)" >> /home/ubuntu/.ssh/authorized_keys' | base64 -w 0) \
+  ...
+```
+This example is in the context of GitHub Actions, hence the `$GITHUB_ACTOR` i.e. the GitHub username for the [person who triggered the workflow](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables).
+
+Now GitHub lists user's Public SSH and GPG keys for convenience/verifcation as an example you can see: [SSH example](https://github.com/casperdcl.keys) or [GPG example](https://github.com/dacbd.gpg).
+
+So this line essentially results in: 
+```
+curl https://github.com/YOURUSERNAME.keys >> ~/.ssh/authorized_keys
+``` 
```

Comment:
I think this is cleaner?

````suggestion
  --cloud-startup-script=$(echo 'echo "$(curl https://github.com/${{ github.actor }}.keys)" >> /home/ubuntu/.ssh/authorized_keys' | base64 -w 0) \
  ...
```

<admon type="info">

GitHub Actions will [replace `${{ github.actor }}` with the username of the person who triggered the workflow](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context). Conveniently, GitHub (and GitLab) provide a URL to access a user's public SSH keys. In effect the above command runs:

```bash
curl https://github.com/YOUR_USERNAME.keys >> ~/.ssh/authorized_keys
```

in the cloud instance.

</admon>

````